### ISSUE TITLE
Inference Speed

### ISSUE THREAD

**Issue by marvision-ai** [Maxfashko/pyretinanet-tensorrt#2]
Hello @Maxfashko 

Great implementation! Love the idea for the repo. Quick question:

What is the inference speed you are getting with the TX2 or Xavier and with what image size? 

Thanks!

**Comment by Maxfashko** [comment]
Hi @mbufi.
TX2: I checked the speed of the inference on the model by resnet50 at the image resolution of 1280 * 800. The speed of the test results ranges from 0.5-0.6 s.
I will get my xavier somewhere in a month.  After that I can make more tests.

**Comment by marvision-ai** [comment]
@Maxfashko With all the tests ive done in the past for networks with regards to tx2 vs. Xavier.. Xavier has always been around 10x faster. So I predict it to be around 50-60 ms/ frame. 

Since you have done such good work with this, I may try it out myself and we can compare!

Thanks again.

**Comment by marvision-ai** [comment]
@Maxfashko I have some updates for you. 

Here are the inference speeds I am getting: 
- **Model you host on this repo** : 60-70 ms/image 
- **Own Model trained with the latest 19.10 branch on the retinanet repo**: 80-90 ms/image

Quick question for you... When you trained your network, was there any changes you made to it to have it run at those speeds? Have you tried fine tuning any network with the latest github changes? I am curious as to why there are such speed discrepancies. If you changed nothing.. then I will have to potentially flag an issue on the main repo due to decreasing inference times. 

Thanks!

**Comment by Maxfashko** [comment]
@mbufi, In the process of training the network, I did not use any optimizations.
If I were you, I would check if the structure of the layers of the neural network has changed.
If the network structure has not changed, then you need to find out what is the difference between docker containers (TPT5 and 19.10 branches)
I don’t remember exactly which container I used (maybe nvcr.io/nvidia/pytorch:19.09-py3)

**Find out whether you use FP16 optimization when converting trained weights**

**Comment by marvision-ai** [comment]
@Maxfashko So I have one more quesiton for you.

So I converted my model to INT8 and it was able to do inference on a 1120x960 image in 30ms: **Using the retinanet cpp infer**

If I then take the same model and use the pyretinanet wrapper I get around 61ms/image.

This github repo implementation is fantastic. I am just curious as to what may be causing the 30ms delay? I tried to dig deeper but was not successful. Before I continue looking, I just wanted to know if you found the same discrepency in performance. (I also understand that there will be delay from c++ to python, but I want to make sure its not my system causing the issues) 

Thanks a bunch!
Cheers.
